pkg/scanner/utils/utils.go

```diff
@@ -101,3 +110,26 @@ func formatVersion(epoch int, version, release string) string {
 	return v
 
 }
+
+func CheckExitCode(exitCode int, results report.Results) {
+	if exitCode != 0 {
+		for _, result := range results {
+			if len(result.Vulnerabilities) > 0 {
+				os.Exit(exitCode)
+			}
+		}
+	}
+}
+
+func FillAndFilterVulns(ctx context.Context, initializeVulnerabilityClient func() vulnerability.Client, results report.Results, severities []dbTypes.Severity, ignoreUnfixed bool, ignoreFile string, ignorePolicy string) error {
+	vulnClient := initializeVulnerabilityClient()
+	for i := range results {
+		vulnClient.FillInfo(results[i].Vulnerabilities, results[i].Type)
+		vulns, err := vulnClient.Filter(ctx, results[i].Vulnerabilities, severities, ignoreUnfixed, ignoreFile, ignorePolicy)
```

Comment:
@knqyf263: Note that this call was missing from client https://github.com/aquasecurity/trivy/pull/642/commits/61f128a299d5dfbfe263fc54d46885bf36d1480e 

Would you know why is that? Unit and integration tests don't seem to be affected by adding it which signals either the lack of assertions or this is done elsewhere in the case of client.